src/canopy_utils_mod.F90

```diff
@@ -251,4 +251,132 @@ real(rk) function GET_GAMMA_CO2(co2_opt, co2_set)  result( GAMMA_CO2 )
+        ! !INTERFACE:
+
+        ! !INPUT PARAMETERS:
+        INTEGER,  INTENT(IN) :: leafage_opt       ! Option for leaf age emission factor calculation
+        ! 0=On;
+        ! 1 or  >1 =off i.e. GAMMA_LEAFAGE =1
+
+        REAl(rk), INTENT(IN) :: tsteplai                      ! time step, number of days between Past and Current LAI inputs
+        REAL(rk), INTENT(IN) :: TABOVE      ! Above canopy temperature (K), t2m or tmpsfc
+        REAL(rk), INTENT(IN) :: LAIpast         ! Past LAI [cm2/cm2]
+        REAL(rk), INTENT(IN) :: LAIcurrent         ! Current LAI [cm2/cm2]
+        REAL(rk), INTENT(IN) :: Anew        ! Relative emiss factor (new leaves)
+        REAL(rk), INTENT(IN) :: Agro        ! Relative emiss factor (growing leaves)
+        REAL(rk), INTENT(IN) :: Amat        ! Relative emiss factor (mature leaves)
+        REAL(rk), INTENT(IN) :: Aold        ! Relative emiss factor (old leaves)
+        !
+        ! !RETURN VALUE:
+        REAL(rk) :: GAMMA_LEAFAGE             ! Leaf age activity factor [unitless]
+        !
+        ! !LOCAL VARIABLES:
+        !INTEGER :: tsteplai                      ! time step
+        REAL(rk) :: Fnew, Fgro                ! foliage fractions
+        REAL(rk) :: Fmat, Fold
+        REAL(rk) :: ti, tm
+
+        !
+        ! !REMARKS:
+        !  The function computes the leaf age activity factor based on BVOC's leaf age response.
+        !
+        ! !REVISION HISTORY:
+        !  Adapted from HEMCO: https://github.com/geoschem/hemco for complete history
+        !EOP
+        !------------------------------------------------------------------------------
+        !BOC
+
+        !-----------------------
+        ! Compute GAMMA_LEAFAGE
+        !-----------------------
+
+
+        !TABOVE (Tt in MEGAN code) -> Above Canopy TEMP (tmp2mref or temp2)
+
+        ! Calculate foliage fraction
+        !-----------------------
+        !Also, Compute ti and tm
+        ! ti: number of days after budbreak required to induce emissions
+        ! tm: number of days after budbreak required to reach peak emissions
+        IF (LAIpast < LAIcurrent) THEN !(i.e. LAI has Increased)
+            IF (TABOVE .le. 303.0_rk) THEN
+                ti = 5.0_rk + 0.7_rk*(300.0_rk-TABOVE)
+            ELSE
+                ti = 2.9_rk
+            ENDIF
+            tm = 2.3_rk*ti
+            !Fnew calculated
+            IF (ti .ge. tsteplai) THEN
+                Fnew = 1.0_rk - (LAIpast/LAIcurrent)
+            ELSE
+                Fnew = (ti/tsteplai) * ( 1.0_rk-(LAIpast/LAIcurrent) )
+            ENDIF
+            !Fmat calculated
+            IF (tm .ge. tsteplai) THEN
+                Fmat = LAIpast/LAIcurrent
+            ELSE
+                Fmat = (LAIpast/LAIcurrent) + ( (tsteplai-tm)/tsteplai ) * ( 1.0_rk-(LAIpast/LAIcurrent) )
+            ENDIF
+
+            Fgro = 1.0_rk - Fnew - Fmat
+            Fold = 0.0_rk
+        ELSEIF (LAIpast == LAIcurrent) THEN !If LAI remains same
```

Comment:
```suggestion
        ELSEIF (abs(LAIpast - LAIcurrent) < 1.0D-5) THEN !If LAI remains same
```

Reply:
@quaz115 As mentioned in my previous comment, we should not compare floating point numbers for exact equality, rather compare with tolerance.

Reply:
@quaz115 Is this resolved, as I don't see any other commits addressing this issue?  Thanks!